particles/Renderer.cpp

```diff
@@ -326,26 +349,22 @@ void Renderer::init(score::gfx::RenderList& renderer)
             cb.setShaderResources(it->second.srb);
             cb.setViewport(QRhiViewport(0, 0, sz.width(), sz.height()));
 
-            assert(this->m_meshBuffer);
-            assert(this->m_meshBuffer->usage().testFlag(QRhiBuffer::VertexBuffer));
+            SCORE_ASSERT(this->m_meshBuffer);
+            SCORE_ASSERT(this->m_meshBuffer->usage().testFlag(QRhiBuffer::VertexBuffer));
 
             const QRhiCommandBuffer::VertexInput bindings[]
                     = {
                 {this->m_meshBuffer, 0},
-                {this->m_meshBuffer, mesh.vertexCount*sizeof(float)},
-                {this->m_particleOffsets, 0},
-                {this->m_particleSpeeds, 0},
-                {this->m_particleControls, 0}
+                {this->m_meshBuffer, m_mesh.vertexCount*sizeof(float)},
+                {this->m_particleOffsets, 0}
             };
 
-            cb.setVertexInput(0, 5, bindings, this->m_idxBuffer, 0, QRhiCommandBuffer::IndexFormat::IndexUInt32);
-
-            mesh.setupBindings(*this->m_meshBuffer, this->m_idxBuffer, cb);
+            cb.setVertexInput(0, 3, bindings, this->m_idxBuffer, 0, QRhiCommandBuffer::IndexFormat::IndexUInt32);
 
             if(this->m_idxBuffer)
-                cb.drawIndexed(mesh.indexCount, instances, 0, mesh.indexCount * 3 * sizeof(float));
+                cb.drawIndexed(m_mesh.indexCount, instances, 0, m_mesh.indexCount * 3 * sizeof(float));
             else
-                cb.draw(mesh.vertexCount, instances, 0, mesh.indexCount * 3 * sizeof(float));
+                cb.draw(m_mesh.vertexCount, instances, 0, 0);
```

Comment:
I think that this code was wrong but it isn't a codepath that we use as we have index buffers.